app/src/lib/stores/app-store.ts

```diff
@@ -1046,11 +1049,8 @@ export class AppStore extends TypedBaseStore<IAppState> {
         // In the case that the aheadBehindCache doesn't have the needed data, try to
         // request it directly from AheadBehindUpdater. This usually happens on initial load
         // before AheadBehindUpdater has run though all the branches.
-        if (
-          aheadBehindOfInferredBranch === null &&
-          this.currentAheadBehindUpdater
-        ) {
-          aheadBehindOfInferredBranch = await this.currentAheadBehindUpdater.executeAsyncTask(
+        if (aheadBehindOfInferredBranch === null) {
+          aheadBehindOfInferredBranch = await aheadBehindUpdater.executeAsyncTask(
```

Comment:
This logic may change slightly once I address the comment in https://github.com/desktop/desktop/pull/9319#discussion_r395810720